util.c

```diff
@@ -5225,6 +5225,32 @@ S_mem_log_common(enum mem_log_type mlt, const UV n,
                 len = 0;
             }
             PERL_UNUSED_RESULT(PerlLIO_write(fd, buf, len));
+#ifdef USE_C_BACKTRACE
+            if(strchr(pmlenv,'c') && (mlt == MLT_NEW_SV)) {
+                /* TODO: get caller package in here when we work out how */
+                len = my_snprintf(buf, sizeof(buf),
+                        "  caller at %s line %d\n",
+                        CopFILE(PL_curcop), CopLINE(PL_curcop));
+                PerlLIO_write(fd, buf, len);
+
+                Perl_c_backtrace *bt = Perl_get_c_backtrace(aTHX_ 3, 3);
+                Perl_c_backtrace_frame *frame;
+                UV i;
+                for (i = 0, frame = bt->frame_info;
+                        i < bt->header.frame_count;
+                        i++, frame++) {
+                    len = my_snprintf(buf, sizeof(buf),
+                            "  frame[%d]: %p %s at %s +0x%lx\n",
+                            i,
+                            frame->addr,
+                            frame->symbol_name_size && frame->symbol_name_offset ? (char *)bt + frame->symbol_name_offset : "-",
+                            frame->object_name_size && frame->object_name_offset ? (char *)bt + frame->object_name_offset : "?",
+                            (char *)frame->addr - (char *)frame->object_base_addr);
+                    PerlLIO_write(fd, buf, len);
```

Comment:
The PerlIO_write() should be wrapped in PERL_UNUSED_RRSULT() to avoid
```
In file included from perl.h:4117,
                 from util.c:26:
iperlsys.h:802:41: warning: ignoring return value of ‘write’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  802 | #  define PerlLIO_write(fd, buf, count) write((fd), (buf), (count))
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
iperlsys.h:802:41: note: in definition of macro ‘PerlLIO_write’
  802 | #  define PerlLIO_write(fd, buf, count) write((fd), (buf), (count))
      |                                         ^~~~~
```